### PR TITLE
fix: unified search provider id

### PR DESCRIPTION
**Pull request by SebastianKrupinski**
## Issue

Unified Search not searching dav properly

Screenshot 2025-01-29 154244

Unified search backend use the provider id to match providers but the front end was using the app id.

This works fine for most search providers as the app id and provider id are the same in most cases (e.g files app; app id is 'files', search provider id is 'files') but in other cases like dav this breaks (e.g. dav app; app id is 'dav' but search provider is 'contacts or events')

Also using the app id is not a good choice as one app can have multiple providers registered (e.g. dav has Contacts, Events and Tasks search) 

## Resolution

Modify front end to use provider id instead of app id

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: core/src/components/UnifiedSearch/UnifiedSearchModal.vue [hunk 1]
```diff
@@ -374,7 +374,7 @@ export default defineComponent({
 			const providersToSearch = this.filteredProviders.length > 0 ? this.filteredProviders : this.providers
 			const searchProvider = (provider, filters) => {
 				const params = {
-					type: provider.appId,
+					type: provider.id,
```
Comment (nfebe) [inline core/src/components/UnifiedSearch/UnifiedSearchModal.vue:377]: We have to do more than this, it's not clear that the providers don't always match with the appIds

**Review by nfebe — CHANGES_REQUESTED** [review]
I would look into this asap

File: core/src/components/UnifiedSearch/UnifiedSearchModal.vue (hunk 1 above)
Comment (SebastianKrupinski) [reply core/src/components/UnifiedSearch/UnifiedSearchModal.vue:377]: Each app can have more then one provider, therefore the provider id is the correct way. The provider id can be customized, for each provider

**Comment by SebastianKrupinski** [comment]
@nfebe your commit broker this again.

image

You can not just make up your own provider id, the id is being sent as 'dav-contacts' which is not valid. 

What issue are you trying to solve maybe I can help you find a proper solution?

**Comment by nfebe** [comment]
> You can not just make up your own provider id, the id is being sent as 'dav-contacts' which is not valid.

Sorry, I converted it to WIP, it's not supposed to use that there a corresponding frontend change which I am adjusting.

File: core/src/components/UnifiedSearch/UnifiedSearchModal.vue (hunk 1 above)
Comment (nfebe) [reply core/src/components/UnifiedSearch/UnifiedSearchModal.vue]: Yeah, however it's possible to have multiple providers with the same ID and this is a problem.

**Comment by nfebe** [comment]
> Also using the app id is not a good choice as one app can have multiple providers registered (e.g. dav has Contacts, Events and Tasks search)

Sure, I added a `searchFrom` property to in `in-folder` search plugin we could use this in the backend, my attempted backend changes where reverted as it would need much more work.

I have added a commit that restores full function with apps like dav while still having in-folder search working as expected.

-----

TODO: 

- [ ] Ensure "load more results" removes other filters

**Review by nfebe — APPROVED** [review]

**Review by provokateurin — APPROVED** [review]
Reminds me of https://github.com/nextcloud/server/pull/47523 :disappointed:
I really dislike when this kind of mess comes to haunt us back way later.
It works almost fine because in most cases it works, but we don't test good enough for "edge cases" (which aren't really edge cases tbh because it's just standard behavior).
